File: C#/TollFeePeriod.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TollFeeCalculator
{
    /**
     * Defines a toll fee period, with start time, non-inclusive end time, and associated fee.
    */
    class TollFeePeriod
    {
        private TimeSpan _StartTime;
        private TimeSpan _NonInclusiveEndTime;
        private int _TollFee;
        /**
         * The fee to associate with the Period
        */
        public int TollFee
        {
            get
            {
                return _TollFee;
            }
        }
        

        /**
         * Default constructor for a toll fee period
         * @param StartTime - the start time of the fee period
         * @param NonInclusiveEndTime - the end time of the fee period, non-inclusive
         * @param TollFee - the toll fee to associate with this period
        */
        public TollFeePeriod(TimeSpan StartTime, TimeSpan NonInclusiveEndTime, int TollFee)
        {
            _StartTime = StartTime;
            _NonInclusiveEndTime = NonInclusiveEndTime;
            _TollFee = TollFee;
        }



        /**
         * Checks if a given DateTime falls within the fee period defined. Is Concerned ONLY with time.
         * @param Date - The Date to check for bounding
         * @return - if the provided date falls within this fee period.
        */
        public bool IsWithinPeriod(DateTime Date)
        {
            return _StartTime >= Date.TimeOfDay && _NonInclusiveEndTime < Date.TimeOfDay;
        }

    }
}

File: src/main/java/com/evolve/services/TollFeePeriod.java
package com.evolve.services;

import java.time.LocalTime;

public class TollFeePeriod {
    private final LocalTime start;
    private final LocalTime end;
    private final int fee;

    public TollFeePeriod(LocalTime start, LocalTime end, int fee) {
        this.start = start;
        this.end = end;
        this.fee = fee;
    }

    public int getFee() {
        return fee;
    }

    /**
     * Check if the given time is within the toll period.
     * @param time  the specified time
     * @return  true if the specified time is within the period, false otherwise.
     */
    public boolean containsTime(LocalTime time) {
        return (start.isBefore(time) || start.equals(time)) && (end.isAfter(time) || end.equals(time));
    }

    @Override
    public String toString() {
        return "TollFeePeriod{" +
                "start=" + start +
                ", end=" + end +
                ", fee=" + fee +
                '}';
    }
}
